### DISCUSSION TITLE
'nan' results for some sources

### DISCUSSION THREAD

**Discussion by MTHGET** [mboquien/cigale#112] [Q&A]
Hi, what could be the reason that I am getting invalid ('nan') results for some of the sources that I am fitting? Most of the sources (85%) in my catalog are fitted but other sources return 'nans' for all the fitted parameters. I have attached my pcigale.ini and an example source that is not fitting.
[test2.txt](https://github.com/user-attachments/files/18188917/test2.txt)

[pcigale.txt](https://github.com/user-attachments/files/18188910/pcigale.txt)

**Comment by mboquien** [comment]
This output is entirely set to NaN when the fit fails. This typically occurs when there is no valid mode (e.g. if all the models are too old of the given z, but I do not think this is the case here), or when the fit is so bad that the likelihood underflows numerically. This can happen for instance when there is a highly discrepant band.

**Reply by MTHGET** [reply]
Thanks for your response. What confuses me is the same sources that are not fitting, they do fit when I use a different code (MAGPHYS).

**Reply by mboquien** [reply]
A possibility is that the parameter space is a little too restrictive. Are the fits reasonably good for the objects that do fit?

**Reply by MTHGET** [reply]
I can't really say the fits are good because there is no convergence in the far-infrared, which is the case for almost all the sources in my catalog. Although I do get the fits, while CIGALE returns 'nan'. I have attached an example SED obtained from MAGPHYS for one of the sources that do not fit with CIGALE.
![331045](https://github.com/user-attachments/assets/99fe3d1d-31ef-4d8c-916c-052905804392)

**Reply by mboquien** [reply]
I do not see how MAGPHYS could return a χ² of only 1.81 with this sort of fit that is off by orders of magnitudes in the IR. I am not surprised that CIGALE would return a NaN in that case. What happens is that the χ² becomes so large that the likelihood underflows to 0 for all the models. I would suggest to check if there is no contamination by another source in the IR that would strongly increase the flux.

**Reply by MTHGET** [reply]
Ok, I see. I will look into it; thank you.

**Reply by MTHGET** [reply]
It turns out that sources that are not fitting have negative fluxes at some bands.

**Reply by mboquien** [reply]
I am unsure how MAGPHYS handles negative fluxes. CIGALE does not do anything special with them, unless they are <-9990, in which case they are just dropped. Other than that, they are considered regular fluxes. You can check in `observations.fits` if that is the case, these are the pre-processed fluxes that CIGALE fits
